src/components/data-list/index.tsx

```diff
@@ -87,17 +76,28 @@ export const DataList = ({
   const [totalRowCount, setTotalRowCount] = useState(0);
   const [pageNumber, setPageNumber] = useState(0);
   const [isImportWindowOpen, setIsImportWindowOpen] = useState(false);
-  const [openExport, setOpenExport] = useState(false);
+  const [openExportFile, setOpenExportFile] = useState(false);
   const [columnVisibilityModel, setColumnVisibilityModel] = useState(
     initialGridState?.columns?.columnVisibilityModel
   );
   const [gridSettingsLoaded, setGridSettingsLoaded] = useState(false);
+  const [gridState, setGridState] = useState(initialGridState);
 
   const whereFilterQuery = getWhereFilterQuery(whereField, whereFieldValue);
   const basicFilterQuery = getBasicFilterQuery(filterLimit, sortColumn, sortOrder, skipLimit);
   const basicExportFilterQuery = getBasicExportFilterQuery(sortColumn, sortOrder);
 
-  const gridSettingsStorageName = `${modelName}DataListSettings`;
+  useEffect(() => {
+    setSearchTerm(searchText);
+  }, [searchText]);
+
+  useEffect(() => {
+    setIsImportWindowOpen(openImport);
+  }, [openImport]);
+
+  useEffect(() => {
+    setOpenExportFile(openExport);
+  }, [openExport]);
 
   useEffect(() => {
```

Comment:
integrated useLocalStorage hook, but still the 'gridSettingsLoaded' check is required to avoid repetitive api calls. I'll further review this with next PR.